ansible_ai_connect/ai/api/wca/api_key_views.py

```diff
@@ -180,6 +180,70 @@ def post(self, request, *args, **kwargs):
 
         return Response(status=HTTP_204_NO_CONTENT)
 
+    @extend_schema(
+        responses={
+            200: OpenApiResponse(description="OK"),
+            401: OpenApiResponse(description="Unauthorized"),
+            403: OpenApiResponse(description="Forbidden"),
+            404: OpenApiResponse(description="Not found"),
+            429: OpenApiResponse(description="Request was throttled"),
+            500: OpenApiResponse(description="Internal service error"),
+        },
+        summary="DELETE WCA key for an Organization",
+        operation_id="wca_api_key_delete",
+    )
+    def delete(self, request, *args, **kwargs):
+        logger.debug("WCA API Key:: DELETE handler")
+
+        organization = None
+        exception = None
+        start_time = time.time()
+        try:
+            organization = request._request.user.organization
+            if not organization:
+                return Response(status=HTTP_400_BAD_REQUEST)
+
+            secret_manager = apps.get_app_config("ai").get_wca_secret_manager()
+            wca_key = secret_manager.get_secret(organization.id, Suffixes.API_KEY)
+            if wca_key is None:
+                return Response(status=HTTP_400_BAD_REQUEST)
+
+            secret_manager.delete_secret(organization.id, Suffixes.API_KEY)
+
+            # Audit trail/logging
+            user_set_wca_api_key.send(
```

Comment:
They've deleted the API Key, so you probably don't want to use `user_set_wca_api_key`.

I'd be tempted to add a new `Signal` so Marty can track deletions if he wishes.

Reply:
I will contact Marty as soon as he comes back. noted this.